### PR TITLE
log-creator and log-auditor working

### PR DESCRIPTION
**Pull request by Schereo**
Both clients are working. Added a tutorial in the README. Usage is similar to summer-school-client.
Extended requirements by the feature to audit multiple logs at once.

### CODE REVIEW

File: clients/frama-c-client/log-auditor/src/get-channel-data/index.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        // eslint-disable-next-line no-prototype-builtins
+        if (hashedFiles.hasOwnProperty(logKey)) {
+            if (hashedFiles[logKey] === payload[logKey]) {
+                console.log('~~~~~~~Log is valid!~~~~~~~');
+                validDocuments++;
+            } else {
+                console.log('~~~~~~~Log is invalid!~~~~~~~');
+            }
+        } else {
+            console.log('The log creator seems to have differnt logs than the auditor.');
```
Comment (dominic22) [inline clients/frama-c-client/log-auditor/src/get-channel-data/index.ts:48]: please fix typo

File: clients/frama-c-client/log-auditor/src/utils/encryption/index.test.ts [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+			public: 'HmvXxyyzaA9B5CMp63xG9ptEkgwmHgaYVStdDsYxzDTX',
+			secret: 'DizmPAxTct7rAxzDXEdSJB4Ni8HRwJ71vrRxLmy3H499'
+		};
+		const prvKey = getHexEncodedKey(keypair.secret);
+		const pubKey = getHexEncodedKey(keypair.public);
+
+		expect(prvKey).toBe('bd0e5f2291ba549f162b3b692738bf711d3d4fd582dcbba1473bab0df81cce64');
+		expect(pubKey).toBe('f93d13cee076a3660d5bd212719654a3c50c9575ae8f19de2c9a0155f3891fea');
+
+		const nonce = 'not a valid nonce since it is too long somaybe host tried to inject a message';
```
Comment (dominic22) [inline clients/frama-c-client/log-auditor/src/utils/encryption/index.test.ts:34]: please fix typo

File: clients/frama-c-client/log-creator/.gitignore [hunk 3]
```diff
@@ -0,0 +1,3 @@
+LogCreatorIdentity.json
+Channel.json
```
Comment (dominic22) [inline clients/frama-c-client/log-creator/.gitignore:2]: this file should be part of the repository please add it for creator and auditor

File: clients/frama-c-client/log-auditor/src/utils/encryption/index.test.ts (hunk 2 above)
Comment (Schereo) [reply clients/frama-c-client/log-auditor/src/utils/encryption/index.test.ts:34]: Fixed it, I just copied this code. So this typo will be present in other clients as well.

File: clients/frama-c-client/log-auditor/src/get-channel-data/index.ts [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        console.log('###########################');
+        auditChannelData(res.data)
+    }
+};
+
+const auditChannelData = async (channelData: any[]) => {
+    if (channelData.length === 0) {
+        return console.log('Data stream is empty');
+    }
+    const hashedFiles = await loadAndHashFiles();
```
Comment (dominic22) [inline clients/frama-c-client/log-auditor/src/get-channel-data/index.ts:25]: isnt this only one file?

File: clients/frama-c-client/log-creator/src/write-stream-channel/index.ts [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+	}
+
+	return hashedFiles;
+};
+
+
+export const writeStream = async (channelAddress: string): Promise<void> => {
+	console.log('Writing logs to stream...')
+	const apiKey = Config.apiKey ? `?api-key=${Config.apiKey}` : '';
+	const hashedFiles = await loadAndHashFiles();
```
Comment (dominic22) [inline clients/frama-c-client/log-creator/src/write-stream-channel/index.ts:29]: isnt this only one file?

File: clients/frama-c-client/log-auditor/src/get-channel-data/index.ts (hunk 4 above)
Comment (dominic22) [reply clients/frama-c-client/log-auditor/src/get-channel-data/index.ts:25]: * it should always log one specific file and write their hash into the channel

File: clients/frama-c-client/log-creator/src/write-stream-channel/index.ts (hunk 5 above)
Comment (dominic22) [reply clients/frama-c-client/log-creator/src/write-stream-channel/index.ts:29]: * it should always log one specific file and write their hash into the channel

**Review by dominic22 — APPROVED** [review]

**Pull request merged**